Raven.Client.Lightweight/Document/RavenClientEnlistment.cs

```diff
@@ -63,13 +62,13 @@ public void Prepare(PreparingEnlistment preparingEnlistment)
 
 			    if (recoveryInformation == null)
 			    {
-			        session.PrepareTransaction(transaction.LocalIdentifier);
+					AsyncPump.Run(() => session.PrepareTransaction(transaction.LocalIdentifier));
```

Comment:
I'll have a look after I got some feedback about the chances that this PR gets accepted